packages/block-library/src/navigation/edit/index.js

```diff
@@ -224,36 +224,54 @@ function Navigation( {
 	// than 1 exists then use the most recent.
 	// The aim is for the block to "just work" from a user perspective using existing data.
 	useEffect( () => {
-		if (
-			hasUncontrolledInnerBlocks ||
-			isCreatingNavigationMenu ||
-			ref ||
-			! navigationMenus?.length
-		) {
+		// Only autofallback to published menus.
+		const fallbackNavigationMenus = navigationMenus?.filter(
+			( menu ) => menu.status === 'publish'
+		);
+
+		if ( hasUncontrolledInnerBlocks || isCreatingNavigationMenu || ref ) {
 			return;
 		}
 
-		navigationMenus.sort( ( menuA, menuB ) => {
-			const menuADate = new Date( menuA.date );
-			const menuBDate = new Date( menuB.date );
-			return menuADate.getTime() < menuBDate.getTime();
-		} );
+		// If there's non fallback navigation menus and
+		// only one classic menu then create a new navigation menu based on it.
+		if ( ! fallbackNavigationMenus?.length && classicMenus?.length === 1 ) {
+			// Only create classic menus once.
+			if ( ! convertingClassicMenu ) {
+				// This is duplicated several times.
+				const onSelectClassicMenu = async ( classicMenu ) => {
+					setConvertingClassicMenu( true );
```

Comment:
I wonder if we could avoid this and use [the conversion status data already exposed by the classic menu hook](https://github.com/WordPress/gutenberg/blob/349240ac3ea4890f0a95b07e4d089b4307a6eab7/packages/block-library/src/navigation/edit/use-convert-classic-menu-to-block-menu.js#L149).

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/44173/commits/1c8f006cc099283a6f32875f38948f9a9685aa74